dali/kernels/signal/resampling.h

```diff
@@ -297,8 +308,16 @@ struct Resampler {
           }
         }
         assert(out_pos >= out_begin && out_pos < out_end);
-        for (int c = 0; c < num_channels; c++)
-          out[out_pos * num_channels + c] = ConvertSatNorm<Out>(tmp[c]);
+        if (downmix) {
+          float out_val = 0;
+          for (int c = 0; c < num_channels; c++)
+            out_val += tmp[c];
+          out_val /= num_channels;
+          out[out_pos] = ConvertSatNorm<Out>(out_val);
+        } else {
+          for (int c = 0; c < num_channels; c++)
+            out[out_pos * num_channels + c] = ConvertSatNorm<Out>(tmp[c]);
+        }
```

Comment:
This is _extremely_ wasteful. We should have some temporary buffer and downmix there, then resample, like we do in the decoder. In the long run that buffer can be smaller and we can downmix in chunks. I'd prefer not to have fused downmixing than have it done this way. Anyway, downmixing is cheap, we can keep it in the decoder.